Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/internal/Snapping/snap.h

```diff
@@ -0,0 +1,406 @@
+    const Point& tp = get(tvpm, vb);
+    FT tol = get(tol_vm, va);
+
+    const FT sq_dist = gt.compute_squared_distance_3_object()(sp, tp);
+    CGAL::Comparison_result res = CGAL::compare(sq_dist, tol * tol);
+
+    if(res != CGAL::LARGER)
+    {
+      typename Vertex_correspondence_map::iterator it =
+        m_vertex_map.insert(std::make_pair(va, vb)).first;
+      const vertex_descriptor vb2 = it->second;
+
+      // If there are multiple candidates for a source vertex, keep the closest target vertex
+      if(vb2 != vb)
+      {
+        typename boost::unordered_map<vertex_descriptor, FT>::iterator dist_it =
+          m_sq_distance_to_snapped_point.find(va);
+        CGAL_assertion(dist_it != m_sq_distance_to_snapped_point.end());
+
+        const FT sq_dist_to_prev_best = dist_it->second;
+        if(CGAL::compare(sq_dist, sq_dist_to_prev_best) == CGAL::SMALLER)
+        {
+          it->second = vb;
+          dist_it->second = sq_dist;
+        }
+      }
+      else
+      {
+        m_sq_distance_to_snapped_point[va] = sq_dist;
+      }
+    }
+  }
+
+private:
+  boost::unordered_map<vertex_descriptor/*source*/, vertex_descriptor/*target*/>& m_vertex_map;
+  boost::unordered_map<vertex_descriptor/*source*/, FT> m_sq_distance_to_snapped_point;
+  const SVPM& svpm;
+  const TVPM& tvpm;
+  const ToleranceMap& tol_vm;
+  const GeomTraits& gt;
+};
+
+// \ingroup PMP_repairing_grp
+//
+// Attempts to snap the border vertices of the source mesh onto the target mesh.
+// A border vertex of the source mesh is snapped to a vertex of the target mesh
+// if its distance to the target mesh vertex is smaller than a user-chosen bound.
+// If any source target vertex can be projected onto multiple vertices of the target
+// mesh, the snapping process is aborted and no vertex is snapped.
+//
+// @tparam PolygonMesh a model of `FaceListGraph` and `MutableFaceGraph`
+// @tparam SourceNamedParameters a sequence of \ref pmp_namedparameters "Named Parameters"
+// @tparam TargetNamedParameters a sequence of \ref pmp_namedparameters "Named Parameters"
+//
+// @param smesh the source mesh whose border vertices might be moved.
+// @param tmesh the target mesh whose vertices are taken as potential new position for the border
+//              vertices of the source mesh.
+// @param snp optional \ref pmp_namedparameters "Named Parameters" related to the source mesh,
+//            amongst those described below:
+//
+// \cgalNamedParamsBegin
+//    \cgalParamBegin{vertex_point_map} the property map with the points associated to the vertices of the source mesh.
+//                                      The type of this map is model of `ReadWritePropertyMap`.
+//                                      If this parameter is omitted, an internal property map for
+//                                      `CGAL::vertex_point_t` must be available in `PolygonMesh`
+//    \cgalParamEnd
+//    BELOW MUST BE DOCUMENTED AS A NAMED PARAMETER
```

Comment:
that's a TODO, right? Actually I'm not sure I would put it as a named parameter. Named parameters are optional, for this function it makes to no sense to call the function without the tolerance.